src/main/scala/mesosphere/marathon/core/task/state/MarathonTaskStatusMapping.scala

```diff
@@ -0,0 +1,32 @@
+package mesosphere.marathon.core.task.state
+
+import org.apache.mesos
+
+/**
+  * To convert `old` mesos.Protos.TaskStatus.TASK_LOST to a proper
+  * MarathonTaskStatus representation (not `Lost` existing), the TaskStatus.Reason is needed. This object
+  * provides the according Reasons for MarathonTaskStatus.Gone, MarathonTaskStatus.Unreachable,
+  * MarathonTaskStatus.Unknown and MarathonTaskStatus.Dropped.
```

Comment:
1. I don't see a set for `Dropped` (iirc it's the fallback when the reason doesn't match any of the defines sets in here)
2. Since the comment nicely describes the relation to `Gone, Unreachable and Unknown`, maybe rename the Sets accordingly? If this would imply naming conflicts, prefix with ReasonXyz?